### PR TITLE
Doc: several improvements on docu of common periph interfaces

### PR DESCRIPTION
**Pull request by smlng**
moved detailed description of some periph interface docu to be visible right at the respective module page in doxygen, similar to periph_gpio.

This was kind of trigged by #7133 ...

### CODE REVIEW

File: drivers/include/periph/i2c.h [hunk 1]
```diff
@@ -11,9 +11,17 @@
...
+ * MCU's I2C bus and periphicals. This interface is intentionally designed as
+ * simple as possible, to allow for easy implementation and maximal portability.
+ *
+ * @note        The current version of this interface only supports the
+ *              7-bit addressing mode.
+ *
+ * @note        This interface is due for remodeling, hence API changes are to
+ *              be expected for upcoming releases.
+ *
+ * @section     sec_debug A note on pull-up resistores for SDA/SCL
```
Comment (gebart) [inline drivers/include/periph/i2c.h]: Resistors

**Comment by gebart** [comment]
You can keep the commits separate, but amend the typo fix to the i2c commit

File: drivers/include/periph/i2c.h [hunk 2]
```diff
@@ -11,12 +11,20 @@
...
+ * MCU's I2C bus and peripherals. This interface is intentionally designed as
+ * simple as possible, to allow for easy implementation and maximal portability.
+ *
+ * @note        The current version of this interface only supports the
+ *              7-bit addressing mode.
+ *
+ * @note        This interface is due for remodeling, hence API changes are to
+ *              be expected for upcoming releases.
+ *
+ * @section     sec_debug A note on pull-up resistors for SDA/SCL
```
Comment (haukepetersen) [inline drivers/include/periph/i2c.h]: If I am not mistaken, the convention is to use markdown sections `# A note on...` in the doxygen, right @miri64?!

Comment (smlng) [inline drivers/include/periph/i2c.h]: I think you're right ... 

to my defense, I copied from [pwm.h](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/periph/pwm.h#L41). Should I fix this anyway (and pwm, too)??

Comment (miri64) [inline drivers/include/periph/i2c.h]: I would be for fixing i2c.h here and fixing pwm.h in a separate PR.

Comment (smlng) [inline drivers/include/periph/i2c.h]: as this RP is about fixing and enhancing the periph docu, don't you think an extra commit for pwm.h would also be fine here?

Comment (miri64) [inline drivers/include/periph/i2c.h]: Oh right, I'm sorry. I thought this PR was just about I²C. Carry on :-)

Comment (smlng) [inline drivers/include/periph/i2c.h]: adapted `i2c.h` (squashed directly) and `pwm.h` to markdown sections as suggested

**Comment by smlng** [comment]
@haukepetersen are we good here?

**Comment by cgundogan** [comment]
ACK and GO

**Pull request merged**